src/core/inductive.ml

```diff
@@ -175,31 +247,41 @@ let gen_rec_type : Sig_state.t -> popt -> sym -> sym list -> term =
     let f_not_rec : tbox -> tvar -> tbox = fun a _ -> a in
     let init : tbox = _Type in
     fold_cons_typ pos codom inj_var build_rec_hyp domrec dom ind_sym
-      cons_sym f_rec f_not_rec init x_str
+      cons_sym f_rec f_not_rec init x_str assoc_predicates
   in
-
-  (* STEP 3: Create the conclusion of the inductive principle *)
-  let codom ts =
-    let x = Bindlib.new_var mkfree x_str in
-    let t = Bindlib.bind_var x (prf_of_p ts (_Vari x)) in
-    _Prod (fapp ind_sym ts) t
+  (* Very close to the function "fold_right2", but "f" is the function
+     "fold_right" and it needs a neutral element. *)
+  let rec clauses ind_list e : tbox = match ind_list with
+    | [] -> e
+    | (i,cl)::q ->
+        let res = clauses q e in
+        List.fold_right (fun a b -> _Impl (case_of i a) b) cl res
   in
-  let conclusion = gen_ind_typ_codom pos ind_sym codom x_str in
+  let t = List.map (clauses ind_list) conclusion_list in
```

Comment:
thanks to use a better name than t for a list

Reply:
`res` is better?